src/fsharp/TastPickle.fs

```diff
@@ -943,7 +948,10 @@ let u_ILAssemblyRef st =
     match tag with
     | 0 ->
         let a, b, c, d, e, f = u_tup6 u_string (u_option u_bytes) (u_option u_ILPublicKey) u_bool (u_option u_ILVersion) (u_option u_string) st
-        ILAssemblyRef.Create(a, b, c, d, e, f)
+        let aref = ILAssemblyRef.Create(a, b, c, d, e, f)
+        // Remap the assembly so we unpickle with an assembly that is equivelant.
+        // The analogous IL importer does its own version of remapping; thus it doesn't require to do this.  
+        st.iilg.RemapAssemblyRef aref
```

Comment:
I'd rather not perform a remap here, but it's to avoid having to use `EqualsIngoringVersion` as part of the built-in type checks.